application/src/main/java/org/togetherjava/tjbot/Application.java

```diff
@@ -67,4 +69,8 @@ public static void runBot(String token, Path databasePath) {
             logger.error("Failed to create database", e);
         }
     }
+
+    private static void onShutdown() {
```

Comment:
I wouldn't make this static. Also, maybe close the database inside it?

Reply:
It had to be static cause the `runBot` is also `static`. I agree to the database thing though, and this will basically lead to the class not being static anymore, so that issue is also solved 👍

Reply:
After some research, I figured out that the database is correctly shutting down its connections already automatically. See https://github.com/jOOQ/jOOQ/issues/10512

Reply:
We can close the currently anonymous `connection` though.

Reply:
Seems `jda.shutdown()` and `jda.shutdownNow()`  takes too long for a shutdown hook.

Reply:
I can not find any nice solution to this. It is getting more and more complex with no clean end in sight.

At this point, it is not worth the little benefit. I do not see anyone using `Application` outside of the context of `main` anyways. So we do not have to offer a graceful shutdown. So the forced kill from outside is everything we need. And resource-closing during a JVM shutdown neither makes sense (they are released due to the JVM process dying anyways) nor is there time for anything that takes longer than a few milliseconds.

Reply:
If you can agree to this, I would like to dismiss your change request.

Reply:
@illuminator3 ?